### PR TITLE
Handing certificates

### PR DESCRIPTION
**Pull request by meletios**
Moving certificates from products to the new agreements

### CODE REVIEW

File: src/main/java/com/hedvig/contract/apartments/query/dao/InsuranceCertificate.kt [hunk 1]
```diff
@@ -10,6 +10,6 @@ class InsuranceCertificate(
 ) {
   companion object {
     fun from(insuranceCertificatePdf: InsuranceCertificatePdf) =
-      InsuranceCertificate(insuranceCertificatePdf.bucketKey, insuranceCertificatePdf.filename)
+      InsuranceCertificate(insuranceCertificatePdf.key, insuranceCertificatePdf.bucketName)
```
Comment (palmenhq) [inline src/main/java/com/hedvig/contract/apartments/query/dao/InsuranceCertificate.kt]: isn't it the reverse order?

File: src/main/java/com/hedvig/contract/apartments/translators/ProductToContractEventListener.kt [hunk 2]
```diff
@@ -202,6 +205,46 @@ class ProductToContractEventListener(
...
+      val contractId = fetchOrCreateContractId(e.memberId)
+
+      val latestCompatAgreement = getLatestCompatAgreement(e.memberId, timeOfKnowledge)
+
+      if (latestCompatAgreement != null
+        && !latestCompatAgreement.skipAgreement
+        && latestCompatAgreement.certificateKey != null
+        && latestCompatAgreement.certificateBucketName != null
+      ) {
+        commandBus.sendAndWait<UpdateInsuranceCertificate>(
```
Comment (palmenhq) [inline src/main/java/com/hedvig/contract/apartments/translators/ProductToContractEventListener.kt]: I believe this should be `sendAndWait<Void>(...`

File: src/main/java/com/hedvig/contract/apartments/translators/query/ConversionErrorRepository.kt [hunk 3]
```diff
@@ -1,5 +1,6 @@
 package com.hedvig.contract.apartments.translators.query
 
+import com.hedvig.contract.apartments.translators.ConversionError
```
Comment (palmenhq) [inline src/main/java/com/hedvig/contract/apartments/translators/query/ConversionErrorRepository.kt]: ?

File: src/main/java/com/hedvig/contract/util/CertificateCannotUpdatedException.kt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package com.hedvig.contract.util
+
+import java.lang.RuntimeException
+
+class CertificateCannotUpdatedException(message: String) : RuntimeException(message)
```
Comment (palmenhq) [inline src/main/java/com/hedvig/contract/util/CertificateCannotUpdatedException.kt]: "CertificateCannot**Be**UpdatedException" maybe?

File: src/main/java/com/hedvig/contract/apartments/translators/query/ConversionErrorRepository.kt (hunk 3 above)
Comment (meletios) [reply src/main/java/com/hedvig/contract/apartments/translators/query/ConversionErrorRepository.kt]: 🤔

File: src/main/java/com/hedvig/contract/util/CertificateCannotUpdatedException.kt (hunk 4 above)
Comment (meletios) [reply src/main/java/com/hedvig/contract/util/CertificateCannotUpdatedException.kt]: Correcto

**Pull request merged**